llama_stack/providers/impls/meta_reference/eval/eval.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+from llama_models.llama3.api.datatypes import *  # noqa: F403
+
+from llama_stack.apis.eval import *  # noqa: F403
+from llama_stack.apis.common.job_types import Job
+from llama_stack.apis.datasetio import DatasetIO
+from llama_stack.apis.datasets import Datasets
+from llama_stack.apis.inference import Inference
+from llama_stack.apis.scoring import Scoring
+
+from .config import MetaReferenceEvalConfig
+
+
+class MetaReferenceEvalImpl(Eval):
+    def __init__(
+        self,
+        config: MetaReferenceEvalConfig,
+        datasetio_api: DatasetIO,
+        datasets_api: Datasets,
+        scoring_api: Scoring,
+        inference_api: Inference,
+    ) -> None:
+        self.config = config
+        self.datasetio_api = datasetio_api
+        self.datasets_api = datasets_api
+        self.scoring_api = scoring_api
+        self.inference_api = inference_api
+
+        # TODO: assume sync job, will need jobs API for async scheduling
+        self.jobs = {}
+
+    async def initialize(self) -> None: ...
+
+    async def shutdown(self) -> None: ...
+
+    async def validate_eval_input_dataset_schema(self, dataset_id: str) -> None:
+        dataset_def = await self.datasets_api.get_dataset(dataset_identifier=dataset_id)
+        if not dataset_def.dataset_schema or len(dataset_def.dataset_schema) == 0:
+            raise ValueError(
+                f"Dataset {dataset_id} does not have a schema defined. Please define a schema for the dataset."
+            )
+
+        # TODO: we will require user defined message types for ToolResponseMessage or include message.context
+        # for now uses basic schema where messages={type: "user", content: "input_query"}
+        for required_column in ["expected_answer", "input_query"]:
```

Comment:
I think we need the `input_query` to be either:

- inputs which can go to a `completion()` method -- so `content: InterleavedTextMedia`
- or which goes to a `chat_completion()` method -- so `messages: List[Message]` 
- or which goes to a `create_agentic_turn()` method -- so whatever part of AgentCreateTurnRequest we want to be controllable at the "eval row" level

Reply:
Agreed! Was thinking we should have separate column schemas to differentiate to decide b/w whether to call `completion`: E.g.

- `chat_completion_input` for `messages: List[Message]`
- `completion_input` of type `content: InterleavedTextMedia`